shared/source/BluetoothPlatform.cpp

```diff
@@ -0,0 +1,218 @@
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+#include <BluetoothPlatform.h>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// BluetoothPlatform
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+BluetoothPlatform::EventHandler *BluetoothPlatform::getEventHandler()
+{
+    return _eventHandler;
+}
+
+void BluetoothPlatform::setEventHandler(BluetoothPlatform::EventHandler *eh)
+{
+    _eventHandler = eh;
```

Comment:
We can probably check for nullptr and use a static empty instance of `BluetoothPlatform::EventHandler` if the user passes nullptr so we avoid checking before every call: 

```c++
BluetoothPlatform::EventHandler BluetoothPlatform::_default_handler;

void BluetoothPlatform::setEventHandler(BluetoothPlatform::EventHandler *eh)
{
   if (eh == nullptr) { 
       _eventHandler = _default_handler;
   } else { 
       _eventHandler = eh;
   }
}
```